components/layout/block.rs

```diff
@@ -1300,7 +1300,12 @@ impl BlockFlow {
             self.base.position.start.b = solution.block_start + self.fragment.margin.block_start
         }
 
-        let block_size = solution.block_size + self.fragment.border_padding.block_start_end();
+        let block_size = if self.fragment.is_replaced() {
```

Comment:
Seems like it'd be cleaner to not include the padding in the block size in the first time? Looks like this is done in `assign_replaced_inline_size_if_necessary` (though it probably should be done in `assign_replaced_block_size_if_necessary`?).

Reply:
I agree at the moment it looks more like fixing the cause rather than the effect.

I'm not sure if I understand though, from those functions I'm guessing you mean change the block size set in border_box (the value that is passed to solve_vertical_constraints_abs_replaced as the block size).  I was afraid to mess with the value of border_box as its doc string said it should contain the padding and border.

```
The position of this fragment relative to its owning flow. The size includes padding and border, but not margin.
```

I tried to find a way to pass the something other than border_box value on https://github.com/bobthekingofegypt/servo/blob/70097503396aab585cc565aaeaa693cce4757800/components/layout/block.rs#L1262 but I don't have enough knowledge of layout to figure out one.  I could adjust the value directly on line 1262 I guess but it's a similar confusion of just removing padding and borders to re-add them a few lines later.

Reply:
Err, you're totally right, I guess we should substract the padding in solve_vertical_constraints_abs_replaced? @mbrubeck, could you take a look? You reviewed those functions when they were added, and I should also read a bit to see what's the right thing here :)

Reply:
Ugh, this code is all a bit of a mess, isn't it?  Umm, maybe this is a better place to subtract the padding:

https://github.com/servo/servo/blob/5c7e79e91c12d19830fdb4c6c17be2a551f88aa2/components/layout/block.rs#L1262

(See also the TODO comments above this line which hint at some of the bigger issues here.)